parallel-consumer-core/src/test/java/io/confluent/parallelconsumer/ParrallelEoSStreamProcessorPauseResumeTest.java

```diff
@@ -0,0 +1,246 @@
+
+        // register unique ID on the parallel consumer
+        String myId = "p/r-test-" + MY_ID_GENERATOR.incrementAndGet();
+        parallelConsumer.setMyId(Optional.of(myId));
+
+        // setup the log capture to be able to follow the two loops
+        LoggerContext lc = (LoggerContext) LoggerFactory.getILoggerFactory();
+
+        // add filter to the STDOUT appender to ensure it stays on level INFO
+        LevelFilter levelInfoFilter = new LevelFilter();
+        levelInfoFilter.setContext(lc);
+        levelInfoFilter.setLevel(Level.INFO);
+        levelInfoFilter.setOnMismatch(FilterReply.DENY);
+        levelInfoFilter.start();
+        lc.getLogger(Logger.ROOT_LOGGER_NAME).getAppender("STDOUT").addFilter(levelInfoFilter);
+
+        // create control loop tracker and register it as log appender
+        controlLoopTracker = new ControlLoopTracker(myId);
+        controlLoopTracker.setContext(lc);
+        Logger parallelStreamProcessorLogger = (Logger) LoggerFactory.getLogger(AbstractParallelEoSStreamProcessor.class);
+        parallelStreamProcessorLogger.setLevel(Level.TRACE);
+        parallelStreamProcessorLogger.addAppender(controlLoopTracker);
+        Logger brokerPollSystemLogger = (Logger) LoggerFactory.getLogger(BrokerPollSystem.class);
+        brokerPollSystemLogger.setLevel(Level.TRACE);
+        brokerPollSystemLogger.addAppender(controlLoopTracker);
+        controlLoopTracker.start();
+    }
+
+    @ParameterizedTest()
+    @EnumSource(CommitMode.class)
+    @SneakyThrows
+    void pausingAndResumingProcessingShouldWork(final CommitMode commitMode) {
+        int numTestRecordsPerBatch = 1_000;
+
+        // setup parallel consumer with custom processing function
+        setupParallelConsumerInstanceAndLogCapture(commitMode);
+        TestUserFunction testUserFunction = new TestUserFunction();
+        parallelConsumer.poll(testUserFunction);
+
+        // produce some messages
+        addRecords(numTestRecordsPerBatch);
+
+        // wait for processing to finish
+        Awaitility
+            .waitAtMost(5L, TimeUnit.SECONDS)
+            .pollDelay(50L, TimeUnit.MILLISECONDS)
+            .until(testUserFunction.numProcessedRecords::get, numRecords -> numTestRecordsPerBatch == numRecords);
+        testUserFunction.reset();
+
+        // pause parallel consumer and wait for control loops to catch up
+        parallelConsumer.pauseIfRunning();
+        controlLoopTracker.waitForSomeParallelStreamProcessorControlLoopCycles(1, 5L, TimeUnit.SECONDS);
+        controlLoopTracker.waitForSomeBrokerPollSystemControlLoopCycles(1, 5L, TimeUnit.SECONDS);
+
+        // produce more messages -> nothing should be processed
+        addRecords(numTestRecordsPerBatch);
+        controlLoopTracker.waitForSomeControlLoopCycles(5, 5L, TimeUnit.SECONDS);
+
+        // shouldn't have produced any records
+        BDDAssertions.assertThat(testUserFunction.numProcessedRecords).hasValue(0L);
```

Comment:
use google Truth instead of assertJ (project is migrating)

Reply:
Replaced AssertJ by Truth in `ParallelEoSStreamProcessorPauseResumeTest`.